backend/app/controllers/spree/admin/cancellations_controller.rb

```diff
@@ -10,8 +10,8 @@ def index
       end
 
       def short_ship
-        inventory_unit_ids = params[:inventory_unit_ids] || []
-        inventory_units = Spree::InventoryUnit.where(id: inventory_unit_ids)
+        inventory_unit_ids = (params[:inventory_unit_ids] || []).map(&:to_i)
+        inventory_units = @order.inventory_units.select { |iu| inventory_unit_ids.include? iu.id }
```

Comment:
@spaghetticode unfortunately no, we can't use `where`: the whole point of this PR is about removing `where` with `select`, since the former doesn't return the same object ids of `@order.inventory_units`, while the latter does